plugin-script/src/main/java/io/kestra/plugin/scripts/exec/scripts/runners/Commands.java

```diff
@@ -0,0 +1,146 @@
+        this.runContext = runContext;
+
+        this.workingDirectory = runContext.tempDir();
+        this.outputDirectory = this.workingDirectory.resolve(IdUtils.create());
+        //noinspection ResultOfMethodCallIgnored
+        this.outputDirectory.toFile().mkdirs();
+
+        this.additionalVars = new HashMap<>(Map.of(
+            "workingDir", workingDirectory.toAbsolutePath().toString(),
+            "outputDir", outputDirectory.toString()
+        ));
+
+        this.logSupplier = LogService.defaultLogSupplier(runContext);
+
+        this.dockerOptions = defaultDockerOptions;
+    }
+
+    public Commands withCommands(List<String> commands) throws IOException, IllegalVariableEvaluationException {
+        return new Commands(
+            runContext,
+            workingDirectory,
+            outputDirectory,
+            additionalVars,
+            ScriptService.uploadInputFiles(runContext, runContext.render(commands, this.additionalVars)),
+            env,
+            logSupplier,
+            runnerType,
+            dockerOptions,
+            warningOnStdErr
+        );
+    }
+
+    public Commands withEnv(Map<String, String> envs) throws IllegalVariableEvaluationException {
+        return new Commands(
+            runContext,
+            workingDirectory,
+            outputDirectory,
+            additionalVars,
+            commands,
+            (envs == null ? Map.<String, String>of() : envs)
+                .entrySet()
+                .stream()
+                .map(throwFunction(r -> new AbstractMap.SimpleEntry<>(
+                        runContext.render(r.getKey(), additionalVars),
+                        runContext.render(r.getValue(), additionalVars)
+                    )
+                ))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)),
+            logSupplier,
+            runnerType,
+            dockerOptions,
+            warningOnStdErr
+        );
+    }
+
+    public Commands addAdditionalVars(Map<String, Object> additionalVars) {
+        this.additionalVars.putAll(additionalVars);
+
+        return this;
+    }
+
+    public Commands addEnv(Map<String, String> envs) {
+        this.env.putAll(envs);
+
+        return this;
+    }
+
+    public ScriptOutput run() throws Exception {
+        RunnerResult runnerResult;
+
+        if (runnerType.equals(RunnerType.DOCKER)) {
+            runnerResult = new DockerScriptRunner(runContext.getApplicationContext()).run(this, this.dockerOptions);
+        } else {
+            runnerResult = new ProcessBuilderScriptRunner().run(this);
+        }
+
+        Map<String, URI> outputFiles = ScriptService.uploadOutputFiles(runContext, outputDirectory);
+
+        Map<String, Object> outputsVars = new HashMap<>();
+        outputsVars.putAll(runnerResult.getStdOut().getOutputs());
+        outputsVars.putAll(runnerResult.getStdErr().getOutputs());
```

Comment:
It's not the full logs, only captured [outputs](https://github.com/kestra-io/plugin-scripts/blob/a4d9a1ab10e36b186ffda335e3ca7d386f027c5a/plugin-script/src/main/java/io/kestra/plugin/scripts/exec/scripts/runners/LogThread.java#L31)

Reply:
This is the logs that the user are writting for ex with echo on Shell right ?
So this is the full log and should not be added to the outputs, the Log task didn't add the message to the output, the same should be done here.
Is there a strong reason why it is added to the output ? If not I strongly advise to not add it to the output as it can be added later if needed by hard to remove it.
Another solution would be to add it inside the internal storage but, if it is added as log, it's already available in the Logs tab of the execution and can be retrieved with the log Fetch task so it is not mandatory to add it to outputs.

Reply:
No it's only capture output with the special syntax `::{...}::`

Reply:
OK, didn't know that, it should be proprely documented

Reply:
https://kestra.io/docs/plugin-developer-guide/outputs maybe should be move on developer documentation